### PR TITLE
Shorter hostnames for live ingresses

### PR DESCRIPTION
**Pull request by razvan-moj**
**Why**
We don't want the main monitoring/login page address to change when we switch clusters

### CODE REVIEW

File: terraform/cloud-platform-components/templates/oauth2-proxy.yaml.tpl [hunk 1]
```diff
@@ -29,7 +29,7 @@ ingress:
   enabled: true
   path: /
   hosts:
-    - "${hostname}"
+    - "${hostname}"
```
Comment (jasonBirchall) [inline terraform/cloud-platform-components/templates/oauth2-proxy.yaml.tpl]: Remove space.

Comment (jasonBirchall) [reply terraform/cloud-platform-components/templates/oauth2-proxy.yaml.tpl]: Done in cb4f98b.

File: terraform/cloud-platform-components/nginx-ingress-acme.tf [hunk 2]
```diff
@@ -30,7 +30,7 @@ controller:
 
   service:
     annotations:
-      external-dns.alpha.kubernetes.io/hostname: "*.apps.${data.terraform_remote_state.cluster.cluster_domain_name},apps.${data.terraform_remote_state.cluster.cluster_domain_name}"
+      external-dns.alpha.kubernetes.io/hostname: "${terraform.workspace == local.live_workspace ? format("*.%s, *.apps.%s, apps.%s", local.live_domain, data.terraform_remote_state.cluster.cluster_domain_name, data.terraform_remote_state.cluster.cluster_domain_name) : format("*.apps.%s, apps.%s", data.terraform_remote_state.cluster.cluster_domain_name, data.terraform_remote_state.cluster.cluster_domain_name)}"
```
Comment (alkar) [inline terraform/cloud-platform-components/nginx-ingress-acme.tf]: I find this a bit hard to parse, can't we instead do something like:
```
"*.apps.${data.terraform_remote_state.cluster.cluster_domain_name},apps.${data.terraform_remote_state.cluster.cluster_domain_name}${terraform.workspace == local.live_workspace ? format(",*.%s", local.live_domain) : ""}"
```

File: terraform/cloud-platform-components/nginx-ingress-acme.tf [hunk 3]
```diff
@@ -59,6 +59,7 @@ data "template_file" "nginx_ingress_default_certificate" {
 
   vars {
     common_name = "*.apps.${data.terraform_remote_state.cluster.cluster_domain_name}"
+    alt_name    = "${terraform.workspace == local.live_workspace ? format("- *.%s", local.live_domain) : ""}"
```
Comment (alkar) [inline terraform/cloud-platform-components/nginx-ingress-acme.tf]: Is that `-` correct?

File: terraform/cloud-platform-components/external-dns.tf [hunk 4]
```diff
@@ -16,8 +16,12 @@ resource "aws_iam_role" "external_dns" {
 
 data "aws_iam_policy_document" "external_dns" {
   statement {
-    actions   = ["route53:ChangeResourceRecordSets"]
-    resources = ["arn:aws:route53:::hostedzone/${data.terraform_remote_state.cluster.hosted_zone_id}"]
+    actions = ["route53:ChangeResourceRecordSets"]
+
+    resources = [
+      "arn:aws:route53:::hostedzone/${data.terraform_remote_state.cluster.hosted_zone_id}",
```
Comment (alkar) [inline terraform/cloud-platform-components/external-dns.tf]: When that ternary is false, this will contain the same zone id twice. Can we not use an empty string for false?

File: terraform/cloud-platform-components/fluentd.tf [hunk 5]
```diff
@@ -5,12 +5,12 @@ resource "helm_release" "fluentd_es" {
 
   set {
     name  = "fluent_elasticsearch_host"
-    value = "${terraform.workspace == local.live_workspace ? "search-cloud-platform-live-dibidbfud3uww3lpxnhj2jdws4.eu-west-2.es.amazonaws.com" : "search-test-yuq2c7ziybjvpmr2tllkghh6va.eu-west-2.es.amazonaws.com"}"
+    value = "${replace(terraform.workspace, "live", "") != terraform.workspace ? "search-cloud-platform-live-dibidbfud3uww3lpxnhj2jdws4.eu-west-2.es.amazonaws.com" : "search-test-yuq2c7ziybjvpmr2tllkghh6va.eu-west-2.es.amazonaws.com"}"
```
Comment (alkar) [inline terraform/cloud-platform-components/fluentd.tf]: Again with this, I think it's a bit hard to parse. Do we currently need to plan for other live clusters? I don't believe we plan to get any more anytime soon. Instead, the previous statement was easier to understand. Wdyt?

File: terraform/cloud-platform-components/nginx-ingress-acme.tf (hunk 3 above)
Comment (razvan-moj) [reply terraform/cloud-platform-components/nginx-ingress-acme.tf]: yes, sadly, indenting :(

File: terraform/cloud-platform-components/external-dns.tf (hunk 4 above)
Comment (razvan-moj) [reply terraform/cloud-platform-components/external-dns.tf]: couldn't find a way, terraform insist on sending an empty pair of quotes that breaks indenting

File: terraform/cloud-platform-components/fluentd.tf (hunk 5 above)
Comment (razvan-moj) [reply terraform/cloud-platform-components/fluentd.tf]: we have live-0 and live-1 so

File: terraform/cloud-platform-components/nginx-ingress-acme.tf (hunk 2 above)
Comment (razvan-moj) [reply terraform/cloud-platform-components/nginx-ingress-acme.tf]: neat! changed

File: terraform/cloud-platform-components/fluentd.tf (hunk 5 above)
Comment (alkar) [reply terraform/cloud-platform-components/fluentd.tf]: I'm saying this because `live-0` is not managed from this branch and won't live long enough to try and integrate. Is `live-2` even a concern right now that would warrant complicating this?

File: terraform/cloud-platform-components/nginx-ingress-acme.tf (hunk 3 above)
Comment (alkar) [reply terraform/cloud-platform-components/nginx-ingress-acme.tf]: I think you missed committing changes to the cert template?

File: terraform/cloud-platform-components/external-dns.tf (hunk 4 above)
Comment (alkar) [reply terraform/cloud-platform-components/external-dns.tf]: Do you mean it breaks validation with the AWS API?
We could do something like this then:
```
["${compact(list(
      "arn:aws:route53:::hostedzone/${data.terraform_remote_state.cluster.hosted_zone_id}",
      "${terraform.workspace == local.live_workspace ? format("%s/%s", "arn:aws:route53:::hostedzone", data.terraform_remote_state.global.cp_zone_id) : ""}",
    ))}"]
```
But I'm not sure this reads better 🍝 
It's just that the fact that the duplicates get pruned when you actually apply is perhaps a "feature" we shouldn't depend upon (couldn't find any mention of this) 🤷‍♂️
